dev/k3d-clean

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+
+k3d cluster list -o json | \
```

Comment:
This looks like an aggressive deletion method, and I wonder if it may be a bit heavy-handed, ie deleting clusters that the user may still want to keep, not necessarily related to Fleet tests. Is this really necessary (just curious)?

How about checking env vars `FLEET_E2E_CLUSTER` and `FLEET_E2E_CLUSTER_DOWNSTREAM` and `k3d cluster delete` them (ie deleting at most 2 clusters)?

Reply:
No, it is not necessary. It was the brute force approach I've taken on my machine without ever having to wonder if anything would still remain. But it does make sense to think more critically about this now.

Since those new scripts don't require having any environment variables set, and because I'd like to keep it that way, I'm even thinking about taking the defaults of single- and multi-cluster environment variables and use those to selectively but always delete **both** kind of clusters, unless environment variables are set. In which case it would depend on the values of those environment variables which clusters would be removed (but it'd still be only up to two, because it would only consider `FLEET_E2E_CLUSTER` and `FLEET_E2E_CLUSTER_DOWNSTREAM`.

What do you think about that?

Reply:
Yes, that makes more sense to me; we'd keep the flexibility of being able to set cluster names while only impacting those when cleaning up test environments.